File: python/extract.py
from bs4 import BeautifulSoup
from glob import glob

output_file = open("extracted.csv","w")
html_files = glob("*.html")
for f in html_files:
   doc = BeautifulSoup(open(f), "lxml")
   rows = doc.find_all("tr",class_="row")
   for row in rows:
      tds = row.find_all("td")
      gamertag = tds[2].next_element.next_element
      points = tds[3].string.replace(",","")
      line = gamertag + "," + points + "\n"
      output_file.write(line)

output_file.close()

File: ruby/extract.rb
require 'nokogiri'

File.open("extracted.csv","w") do |output|
  Dir["*.html"].each do |html_file|
    doc = File.open(html_file) { |f| Nokogiri::HTML(f) }
    doc.css('table tr.row').each do |row|

      name = row.at('./td[3]').text
      points = row.at('./td[4]').text.gsub(",","")

      output.write"#{name},#{points}\n"
    end
  end
end
